src/main/java/com/snowplowanalytics/snowplow/tracker/Tracker.java

```diff
@@ -190,7 +195,98 @@ public TrackerParameters getParameters() {
      * @param event the event to track
      */
     public void track(Event event) {
-        // Emit the event
-        this.emitter.emit(new TrackerEvent(event, this.parameters, this.subject));
+        // a list because Ecommerce events become multiple Payloads
+        List<Event> processedEvents = eventTypeSpecificPreProcessing(event);
```

Comment:
👍 Awesome 👍 
Perhaps you could add the comment directly in the method signature below.
There, you can be more explicit about why we have this weird behaviour as it would be the comment of the method.